src/Umbraco.Web.UI.Client/src/packages/property-editors/checkbox-list/property-editor-ui-checkbox-list.element.ts

```diff
@@ -39,6 +39,12 @@ export class UmbPropertyEditorUICheckboxListElement extends UmbLitElement implem
 							checked: this.#selection.includes(item.value),
 						}));
 		}
+
+
+		this.#selection = this.#selection.filter((selectedItem) =>
+			this._list.some((listItem) => listItem.value === selectedItem),
+		);
+		this.dispatchEvent(new UmbPropertyValueChangeEvent());
```

Comment:
Hi @Zeegaan I would recommend making sure we only fire this change event if the configuration did make a change. I think you could compare length before and after?